rotkehlchen/chain/ethereum/modules/aave/v2/decoder.py

```diff
@@ -25,3 +47,95 @@ def __init__(
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == A_AAVE and
+                event.location_label == to_address and
+                event.balance.amount == amount
+            ):
+                event.event_type = HistoryEventType.STAKING
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Stake {amount} AAVE for {to_address}'
+                event.product = EvmProduct.STAKING
+                event.counterparty = self.counterparty
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == A_STKAAVE and
+                event.location_label == to_address and
+                event.balance.amount == amount
+            ):
+                event.counterparty = self.counterparty
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.notes = f'Receive {event.balance.amount} stkAAVE after staking in AAVE'
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_unstake(
+            self,
+            to_address: ChecksumEvmAddress,
+            amount: FVal,
+            decoded_events: list[EvmEvent],
+    ) -> DecodingOutput:
+        """Decode aave v2 unstake/redeem event"""
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == A_STKAAVE and
+                event.location_label == to_address and
+                event.balance.amount == amount
+            ):
+                event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                event.notes = f'Unstake {amount} stkAAVE for {to_address}'
+                event.product = EvmProduct.STAKING
+                event.counterparty = self.counterparty
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == A_AAVE and
+                event.location_label == to_address and
+                event.balance.amount == amount
+            ):
+                event.counterparty = self.counterparty
+                event.event_type = HistoryEventType.STAKING
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.notes = f'Receive {event.balance.amount} AAVE after unstaking in AAVE'
+
+        return DEFAULT_DECODING_OUTPUT
+
+    # DecoderInterface method
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {
+            STK_AAVE_ADDR: (self._decode_staking_events,),
+            string_to_evm_address(self.pool_address): (self._decode_lending_pool_events,),
```

Comment:
Hmm what are these decode lending pool events? Why are you adding this here?

Reply:
Parent decoders, updated to call super instead.